### ISSUE TITLE
Why Import error?

### ISSUE THREAD

**Issue by ShangwenWang** [Ultimanecat/DefectRepairing#10]
I have already installed unidiff:
![image](https://user-images.githubusercontent.com/59810938/72308770-10aee080-36b8-11ea-9b4a-1ece55cb5073.png)

Then I try to run the command _python3 run.py Chart 1 Patch1_
It is wrong:
![image](https://user-images.githubusercontent.com/59810938/72308828-36d48080-36b8-11ea-95ad-3f6984115622.png)

Why is it?

**Comment by xiongyingfei** [comment]
This error shows that unidiff is not correctly installed. Pls check your installation.

**Comment by ShangwenWang** [comment]
@xiongyingfei Thanks Prof. Xiong.
I have already solved this problem and met another one.
In the README, authors introduce that this experiment was conducted under Defects4j v-1.2. However, in this version, there is no _replaceall-current.jar_ in the folder **test_generation** which is contradict with _Project.pm_ in the **mod-d4j** folder in this repo. As a result, I cannot get correct result. Could you please remember why this happens? Maybe you performed the study under another version of Defects4j like v1.3?

**Comment by Ultimanecat** [comment]
The version we used in the experiment may be a bit older than d4j1.2. It can be an older minor release,. Also, I guess d4j-v1.1 might work. I have no idea whether this script will success on d4j1.3 as we did this experiment before its release. I currently suggest trying d4j-v1.1 if v1.2 fails. We will try to recall the specific version we used in the experiment. Hope this could help you.

**Comment by ShangwenWang** [comment]
@Ultimanecat Thanks for your reply!
However, I guess it must be a version later than v1.2, since in the versions older than v1.2 (like v1.1), the .init.sh files never install _replaceall-current.jar_ file.
Hope you can recall that cause that's really a problem when trying to apply this tool.

**Comment by xinyuan-liu** [comment]
You are right @ShangwenWang . This repo was updated in Aug 2018 with the latest commit of d4j at that time (later than v1.2 but before v1.3 release). I've updated the d4j modification files so it should be competable with d4j v1.3 now.

**Comment by Ringbo** [comment]
@xinyuan-liu Sorry about that, I also meet 2 problems:
1. In the last LIST pass_distance, is it possible the elements be NaN? In the result of PatchRepair5 ,I got the Nan result in  pass_distance list.
such as:

> fali_distance: [7.059298104074019E-4]；
> pass_distance: [NaN,NaN, 0.0, NaN, NaN, NaN, NaN, NaN, NaN, NaN].

2. A uninitialized warning happened in /APR/defects4j/framework/core/Utils.pm line 143. But when I change the tool from randoop to evosuite, it didn't happened again. And the tmp file has been generated nomarlly both in evosuite and randoop.
